### PR TITLE
[Parse] Expand type parsing

### PR DESCRIPTION
**Pull request by rintaro**

### CODE REVIEW

File: Sources/Types.swift [hunk 1]
```diff
@@ -11,8 +11,8 @@ public struct DictionaryType: Type_ {
     public var valueType: Type_
 }
 
-public struct FunctionType­: Type_ {
-    public var arguments: [(String?, Type_)]
+public struct FunctionType: Type_ {
+    public var arguments: [Type_]
```
Comment (rintaro) [inline Sources/Types.swift:15]: @koher Swift compiler doesn't accept labels for arguments in function type representation.
It only accepts `(Int, String) -> Void` or `(_ a: Int, _ b: String) -> Void`
Not `(a: Int, b: String) -> Void`.

Comment (koher) [reply Sources/Types.swift:15]: Oh, I didn't know it. Maybe changed by Swift 3?

LGTM about your changes.

Comment (koher) [reply Sources/Types.swift:15]: `testTupeSuffix` must be a typo 😉

Comment (rintaro) [reply Sources/Types.swift:15]: Yeah, see [SE-0111](https://github.com/apple/swift-evolution/blob/master/proposals/0111-remove-arg-label-type-significance.md) for the rationale.

:flushed: `testTypeSuffix`

**Pull request merged**